flake.nix

```diff
@@ -0,0 +1,38 @@
+{
+  description = "Average flake";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+    crane.url = "github:ipetkov/crane";
+    fenix.url = "github:nix-community/fenix";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = {
+    flake-utils,
+    fenix,
+    nixpkgs,
+    ...
+  } @ inputs:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = nixpkgs.legacyPackages.${system};
+      crane = inputs.crane.mkLib pkgs;
+
+      toolchain = with fenix.packages.${system};
+        combine [
+          minimal.rustc
+          minimal.cargo
+          complete.rustfmt
+          complete.clippy
+        ];
```

Comment:
I usually prefer to have a `rust-toolchain.toml` file and extract from there the information for fenix, but in general I think it is fine